documentation/Software Requirements Specification.md

```diff
@@ -0,0 +1,234 @@
+### Add department
+
+<b>Main scenario:</b>
+* Administrator clicks the “Add” button in the department list view mode;
+* Application displays form to enter department data;
+* Administrator enters department data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
+* If entered data is valid, then record is adding to database;
+* If error occurs, then error message is displaying;
+* If new department record is successfully added, then list of departments with added records is displaying.
+
+<b>Cancel operation scenario:</b>
+* Administrator clicks the “Add” button in the department list view mode;
+* Application displays form to enter department data;
+* Administrator enters department data and presses “Cancel” button;
+* Data don’t save in data base, then list of departments records is displaying to administrator.
+* If the administrator selects the menu item "Department ”or "Employee", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![](add_department.jpg)
+<br>Pic. 2.2 Add department.
+
+<b>When adding a department, the following details are entered:</b>
+* Name department - name department
+* Add employee - adding department employees(optional field)
+### Edit department
+
+<b>Main scenario:</b>
+* Administrator clicks the “Edit” button in the department list view mode;
+* Application displays form to enter department data;
+* Administrator edit department data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
+* If entered data is valid, then edited data is added to database;
+* If error occurs, then error message is displaying;
+* If department record is successfully edited, then list of departments with added records is displaying. 
+
+<b>Cancel operation scenario:</b>
+* Administrator clicks the “Edit” button in the department list view mode;
+* Application displays form to enter department data;
+* Administrator enters department data and presses “Cancel” button;
+* Data don’t save in data base, then list of departments records is displaying to administrator.
+* If the administrator selects the menu item "Department ” or "Employee", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![](edit_department.jpg)
+<br>Pic. 2.3 Edit department.
+
+<b>When editing a order, the following details are entered:</b>
+* Name department - name department
+* Add employee - adding department employees(optional field)
+
+### Removing the department
+
+<b>Main scenario:</b>
+* Administrator, while in the list of departments presses the "Delete" button in the selected department line;
+* If the department can be removed, a confirmation dialog is displayed;
+* Administrator confirms the removal of the order;
+* Record is deleted from database;
+* If error occurs, then error message displays;
```

Comment:
It would be great to see a mockup with error messages.

Reply:
I deleted the line because I didn't remember what kind of error there might be when deleting the field
```python
If error occurs, then error message displays;
```